packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1360,3 +1361,103 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
     properties.add(DiagnosticsProperty<bool>('keepAlive', keepAlive));
   }
 }
+
+// Return a Widget for the given Exception
+Widget _createErrorWidget(Object exception, StackTrace stackTrace) {
+  final FlutterErrorDetails details = FlutterErrorDetails(
+    exception: exception,
+    stack: stackTrace,
+    library: 'widgets library',
+    context: ErrorDescription('building'),
+  );
+  FlutterError.reportError(details);
+  return ErrorWidget.builder(details);
+}
+
+/// Set a cross axis extent constraint to the child Sliver. By default, slivers will try to
+/// fill up the available cross axis extent, but this parent data widget will override that
+/// behavior.
+class SliverConstrainedCrossAxis extends ParentDataWidget<SliverFlexParentData> {
```

Comment:
Nit: dart docs should start with a single sentence followed by a line break.

Just playing with words, I looked at https://api.flutter.dev/flutter/widgets/ConstrainedBox-class.html which is the similar box widget.
Feel free to tweak further. :)

```suggestion
/// Constrains the cross axis extent of the provided sliver to [extent].
///
/// Slivers will typically consume all of the available cross axis extent, but this will impose a constraint
/// on that behavior.
class SliverConstrainedCrossAxis extends ParentDataWidget<SliverFlexParentData> {
```